src/language-js/parse/babel.js

```diff
@@ -163,6 +163,17 @@ function createParse({ isExpression = false, optionsCombinations }) {
     if (isExpression) {
       ast = wrapBabelExpression(ast, opts);
     }
+    // `InterpreterDirective` (hashbang)
+    // Other parsers parse it as comment, babel treat it as comment too
+    // https://github.com/babel/babel/issues/15116
+    else if (ast.program.interpreter) {
+      const {
+        program: { interpreter },
+        comments,
+      } = ast;
+      delete ast.program.interpreter;
+      comments.unshift(interpreter);
+    }
```

Comment:
I think we should do post-process for AST in only `language-js/parse/postprocess/index.js`